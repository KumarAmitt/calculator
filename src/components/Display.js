import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ result }) => (
  <div className="Display">
    <h2 data-testid="result">{result}</h2>
  </div>
);

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
