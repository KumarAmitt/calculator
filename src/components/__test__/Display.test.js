import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Display from '../Display';

test('displays the result', () => {
  const result = '7+31';
  const { getByTestId } = render(<Display result={result} />);
  const resultElement = getByTestId('result');

  expect(resultElement.textContent).toBe(result);
});

test('match snapshot of Display', () => {
  const tree = renderer.create(<Display />).toJSON();

  expect(tree).toMatchSnapshot();
});
