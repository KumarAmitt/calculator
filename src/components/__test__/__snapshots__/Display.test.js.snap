// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot of Display 1`] = `
<div
  className="Display"
>
  <h2
    data-testid="result"
  >
    0
  </h2>
</div>
`;
